<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.302">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Martin Proks, Naz Salehin">
<meta name="dcterms.date" content="2025-01-09">

<title>riveiro_et_al_2023 - Atlas mapping for Redó-Riveiro et al.&nbsp;2023</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_thompson_comparison.html">Reports</a></li><li class="breadcrumb-item"><a href="../notebooks/21_Alba_data_integration.html">Atlas mapping for Redó-Riveiro et al.&nbsp;2023</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">riveiro_et_al_2023</a> 
        <div class="sidebar-tools-main">
    <a href="https://twitter.com/LabBrickman" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/brickmanlab" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../README.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">riveiro_et_al_2023</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_thompson_comparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01 thompson comparison</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/10_re-preprocessing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10 - Re-running preprocessing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/10_re-preprocessing-v2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10 - Re-running preprocessing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/11_figures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/21_Alba_data_integration.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Atlas mapping for Redó-Riveiro et al.&nbsp;2023</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#analysis-with-te" id="toc-analysis-with-te" class="nav-link active" data-scroll-target="#analysis-with-te"><span class="header-section-number">1</span> Analysis with TE</a>
  <ul class="collapse">
  <li><a href="#load-in-experimental-single-cell-rna-seq-data" id="toc-load-in-experimental-single-cell-rna-seq-data" class="nav-link" data-scroll-target="#load-in-experimental-single-cell-rna-seq-data"><span class="header-section-number">1.1</span> Load in experimental single cell RNA-seq data</a></li>
  <li><a href="#load-and-preprocess-the-nowotschin-dataset" id="toc-load-and-preprocess-the-nowotschin-dataset" class="nav-link" data-scroll-target="#load-and-preprocess-the-nowotschin-dataset"><span class="header-section-number">1.2</span> Load and preprocess the Nowotschin dataset</a>
  <ul class="collapse">
  <li><a href="#sanity-plots-for-nowotschin-dataset" id="toc-sanity-plots-for-nowotschin-dataset" class="nav-link" data-scroll-target="#sanity-plots-for-nowotschin-dataset"><span class="header-section-number">1.2.1</span> Sanity plots for Nowotschin dataset</a></li>
  <li><a href="#map-experimental-dataset-onto-altas-nowotschin-dataset" id="toc-map-experimental-dataset-onto-altas-nowotschin-dataset" class="nav-link" data-scroll-target="#map-experimental-dataset-onto-altas-nowotschin-dataset"><span class="header-section-number">1.2.2</span> Map experimental dataset onto altas (Nowotschin) dataset</a></li>
  </ul></li>
  <li><a href="#contingency-table-analysis" id="toc-contingency-table-analysis" class="nav-link" data-scroll-target="#contingency-table-analysis"><span class="header-section-number">1.3</span> Contingency table analysis</a>
  <ul class="collapse">
  <li><a href="#significance-tests-using-clusters" id="toc-significance-tests-using-clusters" class="nav-link" data-scroll-target="#significance-tests-using-clusters"><span class="header-section-number">1.3.1</span> Significance tests using clusters</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#analysis-without-te" id="toc-analysis-without-te" class="nav-link" data-scroll-target="#analysis-without-te"><span class="header-section-number">2</span> Analysis without TE</a>
  <ul class="collapse">
  <li><a href="#load-in-experimental-single-cell-rna-seq-data-1" id="toc-load-in-experimental-single-cell-rna-seq-data-1" class="nav-link" data-scroll-target="#load-in-experimental-single-cell-rna-seq-data-1"><span class="header-section-number">2.1</span> Load in experimental single cell RNA-seq data</a>
  <ul class="collapse">
  <li><a href="#map-experimental-data-onto-filtered-atlas" id="toc-map-experimental-data-onto-filtered-atlas" class="nav-link" data-scroll-target="#map-experimental-data-onto-filtered-atlas"><span class="header-section-number">2.1.1</span> Map experimental data onto filtered atlas</a></li>
  </ul></li>
  <li><a href="#contingency-table-analysis-1" id="toc-contingency-table-analysis-1" class="nav-link" data-scroll-target="#contingency-table-analysis-1"><span class="header-section-number">2.2</span> Contingency table analysis</a>
  <ul class="collapse">
  <li><a href="#significance-tests-using-clusters-1" id="toc-significance-tests-using-clusters-1" class="nav-link" data-scroll-target="#significance-tests-using-clusters-1"><span class="header-section-number">2.2.1</span> Significance tests using clusters</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#session-information" id="toc-session-information" class="nav-link" data-scroll-target="#session-information"><span class="header-section-number">3</span> Session information</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Atlas mapping for Redó-Riveiro et al.&nbsp;2023</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Martin Proks, Naz Salehin </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 9, 2025</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="changelog" class="level4" data-number="0.0.0.1">
<h4 data-number="0.0.0.1" data-anchor-id="changelog"><span class="header-section-number">0.0.0.1</span> Changelog</h4>
<p><strong>25-09-2023</strong></p>
<p>Integration of the dataset without TE</p>
<p><strong>21-09-2023</strong></p>
<p>Analysis with STARsolo based quantification</p>
<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> rich <span class="im">import</span> <span class="bu">print</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> warnings</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># from lightning_fabric.plugins.environments.slurm import PossibleUserWarning</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># warnings.simplefilter(action='ignore', category=PossibleUserWarning)</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(action<span class="op">=</span><span class="st">'ignore'</span>, category<span class="op">=</span><span class="pp">UserWarning</span>)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>warnings.simplefilter(action<span class="op">=</span><span class="st">'ignore'</span>, category<span class="op">=</span><span class="pp">FutureWarning</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>sc.settings.figdir <span class="op">=</span> <span class="st">'../figures/'</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>sc.set_figure_params(dpi<span class="op">=</span><span class="dv">120</span>, dpi_save <span class="op">=</span> <span class="dv">300</span>, <span class="bu">format</span><span class="op">=</span><span class="st">'svg'</span>, transparent<span class="op">=</span><span class="va">True</span>, figsize<span class="op">=</span>(<span class="dv">6</span>,<span class="dv">5</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>sns.set_style(<span class="st">"white"</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.print_figure_kwargs<span class="op">=</span>{<span class="st">'facecolor'</span> : <span class="st">"w"</span>}</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>config InlineBackend.figure_format<span class="op">=</span><span class="st">'retina'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis-with-te" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Analysis with TE</h1>
<section id="load-in-experimental-single-cell-rna-seq-data" class="level2" data-number="1.1">
<h2 data-number="1.1" data-anchor-id="load-in-experimental-single-cell-rna-seq-data"><span class="header-section-number">1.1</span> Load in experimental single cell RNA-seq data</h2>
<div class="cell" data-execution_count="3">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>alba_adata <span class="op">=</span> sc.read(<span class="st">'../data/processed/10_dataset_v2.h5ad'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="4">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(alba_adata, color<span class="op">=</span>[<span class="st">'Stage'</span>, <span class="st">'leiden'</span>, <span class="st">'Sox2'</span>, <span class="st">'Gata6'</span>, <span class="st">'Pou5f1'</span>, <span class="st">'S2G6+'</span>,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                         <span class="st">'All.Events.GFP-A.Geo.Mean'</span>, <span class="st">'All.Events.561D-A.Geo.Mean'</span>], ncols<span class="op">=</span><span class="dv">3</span>, frameon<span class="op">=</span><span class="va">False</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-5-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="5">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>alba_adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="5">
<pre><code>AnnData object with n_obs × n_vars = 1139 × 21590
    obs: 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'n_genes', 'n_counts', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'gene_ids', 'feature_types', 'mt', 'ercc', 'ribo', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells', 'highly_variable', 'means', 'dispersions', 'dispersions_norm', 'highly_variable_nbatches', 'highly_variable_intersection', 'mean', 'std'
    uns: 'Amp_batch_ID_colors', 'S2G6+_colors', 'Stage_colors', 'dendrogram_Stage', 'hvg', 'leiden', 'leiden_colors', 'log1p', 'neighbors', 'pca', 'plate_ID_colors', 'rank_genes_groups', 'umap'
    obsm: 'X_pca', 'X_umap'
    varm: 'PCs'
    layers: 'counts'
    obsp: 'connectivities', 'distances'</code></pre>
</div>
</div>
</section>
<section id="load-and-preprocess-the-nowotschin-dataset" class="level2" data-number="1.2">
<h2 data-number="1.2" data-anchor-id="load-and-preprocess-the-nowotschin-dataset"><span class="header-section-number">1.2</span> Load and preprocess the Nowotschin dataset</h2>
<div class="cell" data-execution_count="6">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> sc.read(<span class="st">"../data/external/Nowotschin_et_al_2019/sc_endoderm_all_cells.h5ad"</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>meta <span class="op">=</span> pd.read_csv(<span class="st">"../data/external/Nowotschin_et_al_2019/e35_cell_types.csv"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>nowo.obs.loc[meta.index, <span class="st">'CellType'</span>] <span class="op">=</span> meta.CellType</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    nowo.obs.Timepoint.isin([<span class="st">'E3.5'</span>, <span class="st">'E4.5'</span>])</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>].copy()</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NOWO_1"</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Nowotschin et al., 2019"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"10X 3' v2"</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'ct'</span>] <span class="op">=</span> nowo.obs[[<span class="st">'Timepoint'</span>, <span class="st">'CellType'</span>]].agg(<span class="st">'-'</span>.join, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'ct_orig'</span>] <span class="op">=</span> nowo.obs.CellType</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>nowo.obs <span class="op">=</span> nowo.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>nowo.var_names <span class="op">=</span> nowo.var_names.<span class="bu">str</span>.lower()</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(nowo, min_counts<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(nowo, min_genes<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>nowo.layers[<span class="st">"counts"</span>] <span class="op">=</span> nowo.X.copy()</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(nowo, target_sum<span class="op">=</span><span class="dv">10_000</span>)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(nowo)</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>nowo.raw <span class="op">=</span> nowo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="7">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>sc.pp.highly_variable_genes(</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>    nowo,</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    flavor<span class="op">=</span><span class="st">"seurat_v3"</span>,</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    n_top_genes<span class="op">=</span><span class="dv">1_000</span>,</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    layer<span class="op">=</span><span class="st">"counts"</span>,</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    batch_key<span class="op">=</span><span class="st">"batch"</span>,</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    min_mean<span class="op">=</span><span class="fl">0.0125</span>, max_mean<span class="op">=</span><span class="dv">10</span>, min_disp<span class="op">=</span><span class="fl">0.5</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    subset<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>sc.pp.pca(nowo)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(nowo)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(nowo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/umap/distances.py:1063: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/umap/distances.py:1071: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/umap/distances.py:1086: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
/projects/dan1/data/Brickman/conda/envs/scvi-1.0.0/lib/python3.10/site-packages/umap/umap_.py:660: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  @numba.jit()
2023-10-23 16:33:03.432403: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-10-23 16:33:03.482766: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-10-23 16:33:04.693881: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT</code></pre>
</div>
</div>
<section id="sanity-plots-for-nowotschin-dataset" class="level3" data-number="1.2.1">
<h3 data-number="1.2.1" data-anchor-id="sanity-plots-for-nowotschin-dataset"><span class="header-section-number">1.2.1</span> Sanity plots for Nowotschin dataset</h3>
<div class="cell" data-execution_count="8">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>    nowo,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">"ct"</span>],</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_01_Nowotschin_only-with_TE-UMAP.svg"</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/umap_integration_01_Nowotschin_only-with_TE-UMAP.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-9-output-2.png"></p>
</div>
</div>
<div class="cell" data-execution_count="9">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>    nowo,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">"ct"</span>],</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    dimensions<span class="op">=</span>[(<span class="dv">0</span>, <span class="dv">1</span>)],</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_02_Nowotschin_only-with_TE-PCA.svg"</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/pca_integration_02_Nowotschin_only-with_TE-PCA.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-10-output-2.png"></p>
</div>
</div>
<div class="cell" data-execution_count="10">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[:, nowo.var.highly_variable].copy()</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>sc.pp.scale(nowo, max_value<span class="op">=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="map-experimental-dataset-onto-altas-nowotschin-dataset" class="level3" data-number="1.2.2">
<h3 data-number="1.2.2" data-anchor-id="map-experimental-dataset-onto-altas-nowotschin-dataset"><span class="header-section-number">1.2.2</span> Map experimental dataset onto altas (Nowotschin) dataset</h3>
<div class="cell" data-execution_count="11">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>nowo.var_names <span class="op">=</span> nowo.var_names.<span class="bu">str</span>.upper()</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>alba_adata.var_names <span class="op">=</span> alba_adata.var_names.<span class="bu">str</span>.upper()</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>var_names <span class="op">=</span> nowo.var_names.intersection(alba_adata.var_names)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[:, var_names].copy()</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>alba_adata <span class="op">=</span> alba_adata[:, var_names].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="12">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>sc.tl.ingest(alba_adata, nowo, obs <span class="op">=</span> <span class="st">'ct'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="13">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>adata_concat <span class="op">=</span> nowo.concatenate(alba_adata, batch_categories<span class="op">=</span>[<span class="st">'nowo'</span>, <span class="st">'alba'</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="14">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>adata_concat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="14">
<pre><code>AnnData object with n_obs × n_vars = 2145 × 919
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'n_counts', 'n_genes', 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'highly_variable-alba', 'means-alba', 'highly_variable_nbatches-alba', 'mean-alba', 'std-alba', 'gene_ids-alba', 'feature_types-alba', 'mt-alba', 'ercc-alba', 'ribo-alba', 'n_cells_by_counts-alba', 'mean_counts-alba', 'pct_dropout_by_counts-alba', 'total_counts-alba', 'n_cells-alba', 'dispersions-alba', 'dispersions_norm-alba', 'highly_variable_intersection-alba', 'highly_variable-nowo', 'highly_variable_rank-nowo', 'means-nowo', 'variances-nowo', 'variances_norm-nowo', 'highly_variable_nbatches-nowo', 'mean-nowo', 'std-nowo'
    obsm: 'X_pca', 'X_umap'
    layers: 'counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>alba_annotated <span class="op">=</span> adata_concat[adata_concat.obs.batch <span class="op">==</span> <span class="st">'alba'</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>alba_annotated</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="16">
<pre><code>View of AnnData object with n_obs × n_vars = 1139 × 919
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'n_counts', 'n_genes', 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'highly_variable-alba', 'means-alba', 'highly_variable_nbatches-alba', 'mean-alba', 'std-alba', 'gene_ids-alba', 'feature_types-alba', 'mt-alba', 'ercc-alba', 'ribo-alba', 'n_cells_by_counts-alba', 'mean_counts-alba', 'pct_dropout_by_counts-alba', 'total_counts-alba', 'n_cells-alba', 'dispersions-alba', 'dispersions_norm-alba', 'highly_variable_intersection-alba', 'highly_variable-nowo', 'highly_variable_rank-nowo', 'means-nowo', 'variances-nowo', 'variances_norm-nowo', 'highly_variable_nbatches-nowo', 'mean-nowo', 'std-nowo'
    obsm: 'X_pca', 'X_umap'
    layers: 'counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="17">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>pd.crosstab(alba_annotated.obs.Stage, alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="17">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">ct</th>
<th data-quarto-table-cell-role="th">E3.5-EPI</th>
<th data-quarto-table-cell-role="th">E3.5-ICM</th>
<th data-quarto-table-cell-role="th">E3.5-PrE</th>
<th data-quarto-table-cell-role="th">E3.5-TE</th>
<th data-quarto-table-cell-role="th">E4.5-EPI</th>
<th data-quarto-table-cell-role="th">E4.5-PrE</th>
<th data-quarto-table-cell-role="th">E4.5-TE</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Stage</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">2iLIF</td>
<td>90</td>
<td>12</td>
<td>45</td>
<td>10</td>
<td>2</td>
<td>1</td>
<td>3</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">EPSC_GFP</td>
<td>90</td>
<td>5</td>
<td>47</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">EPSC_double</td>
<td>87</td>
<td>64</td>
<td>173</td>
<td>2</td>
<td>0</td>
<td>1</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">KOSR_GFP</td>
<td>62</td>
<td>7</td>
<td>63</td>
<td>2</td>
<td>2</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">KOSR_double</td>
<td>75</td>
<td>55</td>
<td>232</td>
<td>7</td>
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>pd.crosstab(alba_adata.obs.Stage, alba_adata.obs.leiden)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="18">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">leiden</th>
<th data-quarto-table-cell-role="th">0</th>
<th data-quarto-table-cell-role="th">1</th>
<th data-quarto-table-cell-role="th">2</th>
<th data-quarto-table-cell-role="th">3</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Stage</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">2iLIF</td>
<td>0</td>
<td>1</td>
<td>0</td>
<td>162</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">EPSC_GFP</td>
<td>1</td>
<td>141</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">EPSC_double</td>
<td>114</td>
<td>213</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">KOSR_GFP</td>
<td>6</td>
<td>0</td>
<td>128</td>
<td>3</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">KOSR_double</td>
<td>310</td>
<td>3</td>
<td>56</td>
<td>1</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="19">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(sc.metrics.confusion_matrix(<span class="st">"Stage"</span>, <span class="st">"leiden"</span>, alba_annotated.obs), cmap<span class="op">=</span>sns.dark_palette(<span class="st">"white"</span>, reverse<span class="op">=</span><span class="va">True</span>, as_cmap<span class="op">=</span><span class="va">True</span>),linewidth<span class="op">=</span><span class="fl">.5</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/crosstab_01_Stage_vs_Leiden_cluster.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-20-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>alba_annotated.obs.Stage.value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="20">
<pre><code>KOSR_double    370
EPSC_double    327
2iLIF          163
EPSC_GFP       142
KOSR_GFP       137
Name: Stage, dtype: int64</code></pre>
</div>
</div>
<div class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>alba_annotated_df<span class="op">=</span>pd.DataFrame({</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'condition'</span>:alba_annotated.obs.Stage,</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'annotation'</span>:alba_annotated.obs.ct,</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="22">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>proportions_df <span class="op">=</span> pd.crosstab(alba_annotated.obs.Stage,alba_annotated.obs.ct, normalize<span class="op">=</span><span class="st">'index'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="23">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>proportions_df_notNorm <span class="op">=</span> pd.crosstab(alba_annotated.obs.Stage,alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="24">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> proportions_df.plot.barh(</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>    stacked<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>    grid<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_03_CellProportions_Integrated_with-TE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-25-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="25">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>proportions_of_clusters_df <span class="op">=</span> pd.crosstab(alba_annotated.obs.leiden, alba_annotated.obs.ct, normalize<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>proportions_of_clusters_df_notNorm <span class="op">=</span> pd.crosstab(alba_annotated.obs.leiden, alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="26">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> proportions_of_clusters_df.plot.barh(</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>    stacked<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>    grid<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_04_CellProportionsLeidenClusters_Integrated_with-TE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-27-output-1.png"></p>
</div>
</div>
</section>
</section>
<section id="contingency-table-analysis" class="level2" data-number="1.3">
<h2 data-number="1.3" data-anchor-id="contingency-table-analysis"><span class="header-section-number">1.3</span> Contingency table analysis</h2>
<div class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scipy.stats <span class="im">as</span> stats</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="28">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>stats.chi2_contingency(proportions_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="28">
<pre><code>Chi2ContingencyResult(statistic=201.78632492668822, pvalue=4.715612624607903e-30, dof=24, expected_freq=array([[ 57.81562774,  20.46444249,  80.1404741 ,   3.00526778,
          0.71553995,   0.28621598,   0.57243196],
       [ 50.36698859,  17.82791923,  69.81562774,   2.61808604,
          0.62335382,   0.24934153,   0.49868306],
       [115.98595259,  41.05443371, 160.77260755,   6.02897278,
          1.43546971,   0.57418788,   1.14837577],
       [ 48.59350307,  17.20017559,  67.35733099,   2.52589991,
          0.60140474,   0.2405619 ,   0.48112379],
       [131.23792801,  46.45302897, 181.91395961,   6.82177349,
          1.62423178,   0.64969271,   1.29938543]]))</code></pre>
</div>
</div>
<div class="cell" data-execution_count="29">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>alba_chi2 <span class="op">=</span> stats.chi2_contingency(proportions_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="30">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>alba_chi2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="30">
<pre><code>Chi2ContingencyResult(statistic=201.78632492668822, pvalue=4.715612624607903e-30, dof=24, expected_freq=array([[ 57.81562774,  20.46444249,  80.1404741 ,   3.00526778,
          0.71553995,   0.28621598,   0.57243196],
       [ 50.36698859,  17.82791923,  69.81562774,   2.61808604,
          0.62335382,   0.24934153,   0.49868306],
       [115.98595259,  41.05443371, 160.77260755,   6.02897278,
          1.43546971,   0.57418788,   1.14837577],
       [ 48.59350307,  17.20017559,  67.35733099,   2.52589991,
          0.60140474,   0.2405619 ,   0.48112379],
       [131.23792801,  46.45302897, 181.91395961,   6.82177349,
          1.62423178,   0.64969271,   1.29938543]]))</code></pre>
</div>
</div>
<div class="cell" data-execution_count="31">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>alba_chi2.pvalue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="31">
<pre><code>4.715612624607903e-30</code></pre>
</div>
</div>
<div class="cell" data-execution_count="32">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>alba_chi2.statistic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="32">
<pre><code>201.78632492668822</code></pre>
</div>
</div>
<div class="cell" data-execution_count="33">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/20453729/what-is-the-equivalent-of-r-data-chisqresiduals-in-python</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.stats.contingency <span class="im">import</span> margins</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> residuals(observed, expected):</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> (observed <span class="op">-</span> expected) <span class="op">/</span> np.sqrt(expected)</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> stdres(observed, expected):</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> observed.<span class="bu">sum</span>()</span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>    rsum, csum <span class="op">=</span> margins(observed)</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># With integers, the calculation</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>    <span class="co">#     csum * rsum * (n - rsum) * (n - csum)</span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># might overflow, so convert rsum and csum to floating point.</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>    rsum <span class="op">=</span> rsum.astype(np.float64)</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>    csum <span class="op">=</span> csum.astype(np.float64)</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>    v <span class="op">=</span> csum <span class="op">*</span> rsum <span class="op">*</span> (n <span class="op">-</span> rsum) <span class="op">*</span> (n <span class="op">-</span> csum) <span class="op">/</span> n<span class="op">**</span><span class="dv">3</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> (observed <span class="op">-</span> expected) <span class="op">/</span> np.sqrt(v)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="34">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>sns.set_style(<span class="st">"dark"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="35">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(stdres(proportions_df_notNorm.to_numpy(), alba_chi2.expected_freq), columns<span class="op">=</span> proportions_df_notNorm.columns, index<span class="op">=</span>proportions_df_notNorm.index),</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_05_ChiSquared_NormalisedResiduals-withTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-36-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="36">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(residuals(proportions_df_notNorm.to_numpy(), alba_chi2.expected_freq), columns<span class="op">=</span> proportions_df_notNorm.columns, index<span class="op">=</span>proportions_df_notNorm.index),</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_06_ChiSquared_RawResiduals-withTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-37-output-1.png"></p>
</div>
</div>
<section id="significance-tests-using-clusters" class="level3" data-number="1.3.1">
<h3 data-number="1.3.1" data-anchor-id="significance-tests-using-clusters"><span class="header-section-number">1.3.1</span> Significance tests using clusters</h3>
<div class="cell" data-execution_count="37">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>stats.chi2_contingency(proportions_of_clusters_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="37">
<pre><code>Chi2ContingencyResult(statistic=196.4951224929756, pvalue=5.025159340654328e-32, dof=18, expected_freq=array([[152.87445127,  54.11150132, 211.90517998,   7.94644425,
          1.89201054,   0.75680421,   1.51360843],
       [126.98156277,  44.94644425, 176.01404741,   6.60052678,
          1.57155399,   0.6286216 ,   1.2572432 ],
       [ 65.2642669 ,  23.10096576,  90.46532046,   3.39244952,
          0.80772608,   0.32309043,   0.64618086],
       [ 58.87971905,  20.84108867,  81.61545215,   3.06057946,
          0.72870939,   0.29148376,   0.58296752]]))</code></pre>
</div>
</div>
<div class="cell" data-execution_count="38">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2 <span class="op">=</span> stats.chi2_contingency(proportions_of_clusters_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="39">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2.pvalue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="39">
<pre><code>5.025159340654328e-32</code></pre>
</div>
</div>
<div class="cell" data-execution_count="40">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2.statistic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="40">
<pre><code>196.4951224929756</code></pre>
</div>
</div>
<div class="cell" data-execution_count="41">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(stdres(proportions_of_clusters_df_notNorm.to_numpy(), seurat_clusters_chi2.expected_freq), columns<span class="op">=</span> proportions_of_clusters_df_notNorm.columns, index<span class="op">=</span>proportions_of_clusters_df_notNorm.index),</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span>, center <span class="op">=</span> <span class="fl">0.0</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_07_LeidenClusters_ChiSquared_NormalisedResiduals-withTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-42-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(residuals(proportions_of_clusters_df_notNorm.to_numpy(), seurat_clusters_chi2.expected_freq), columns<span class="op">=</span> proportions_of_clusters_df_notNorm.columns, index<span class="op">=</span>proportions_of_clusters_df_notNorm.index),</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span>, center <span class="op">=</span> <span class="fl">0.0</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_08_LeidenClusters_ChiSquared_RawResiduals-withTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-43-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata_concat,</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>, <span class="st">'leiden'</span>],</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-44-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(adata_concat,</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>],</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    ncols <span class="op">=</span> <span class="dv">1</span>,</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_09_UMAP_integrated_withTE.svg"</span>,</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/umap_integration_09_UMAP_integrated_withTE.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-45-output-2.png"></p>
</div>
</div>
<div class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata_concat,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>],</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_10_PCA_integrated_withTE.svg"</span>,</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/pca_integration_10_PCA_integrated_withTE.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-46-output-2.png"></p>
</div>
</div>
</section>
</section>
</section>
<section id="analysis-without-te" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Analysis without TE</h1>
<p>It appears the majority of cells map to ICM, EPI, PE labels. The follow up analysis excludes the trophectoderm (TE) populations to allow for greater resolution.</p>
<p>Load in the Nowotschin et al.&nbsp;dataset, but this time, exclude cells labelled as TE</p>
<div class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> sc.read(<span class="st">"../data/external/Nowotschin_et_al_2019/sc_endoderm_all_cells.h5ad"</span>)</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>meta <span class="op">=</span> pd.read_csv(<span class="st">"../data/external/Nowotschin_et_al_2019/e35_cell_types.csv"</span>, index_col<span class="op">=</span><span class="dv">0</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>nowo.obs.loc[meta.index, <span class="st">'CellType'</span>] <span class="op">=</span> meta.CellType</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>    nowo.obs.Timepoint.isin([<span class="st">'E3.5'</span>, <span class="st">'E4.5'</span>]) <span class="op">&amp;</span> nowo.obs.CellType.isin([<span class="st">'EPI'</span>,<span class="st">'ICM'</span>,<span class="st">'PrE'</span>])</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>].copy()</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NOWO_1"</span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Nowotschin et al., 2019"</span></span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"10X 3' v2"</span></span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'ct'</span>] <span class="op">=</span> nowo.obs[[<span class="st">'Timepoint'</span>, <span class="st">'CellType'</span>]].agg(<span class="st">'-'</span>.join, axis<span class="op">=</span><span class="dv">1</span>)</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>nowo.obs[<span class="st">'ct_orig'</span>] <span class="op">=</span> nowo.obs.CellType</span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a>nowo.obs <span class="op">=</span> nowo.obs[[<span class="st">'batch'</span>, <span class="st">'experiment'</span>, <span class="st">'technology'</span>, <span class="st">'ct'</span>, <span class="st">'ct_orig'</span>]]</span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a>nowo.var_names <span class="op">=</span> nowo.var_names.<span class="bu">str</span>.lower()</span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(nowo, min_counts<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>sc.pp.filter_cells(nowo, min_genes<span class="op">=</span><span class="dv">10</span>)</span>
<span id="cb62-17"><a href="#cb62-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-18"><a href="#cb62-18" aria-hidden="true" tabindex="-1"></a>nowo.layers[<span class="st">"counts"</span>] <span class="op">=</span> nowo.X.copy()</span>
<span id="cb62-19"><a href="#cb62-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-20"><a href="#cb62-20" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(nowo, target_sum<span class="op">=</span><span class="dv">10_000</span>)</span>
<span id="cb62-21"><a href="#cb62-21" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(nowo)</span>
<span id="cb62-22"><a href="#cb62-22" aria-hidden="true" tabindex="-1"></a>nowo.raw <span class="op">=</span> nowo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="47">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>sc.pp.highly_variable_genes(</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>    nowo,</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>    flavor<span class="op">=</span><span class="st">"seurat_v3"</span>,</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>    n_top_genes<span class="op">=</span><span class="dv">1_000</span>,</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>    layer<span class="op">=</span><span class="st">"counts"</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>    batch_key<span class="op">=</span><span class="st">"batch"</span>,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>    min_mean<span class="op">=</span><span class="fl">0.0125</span>, max_mean<span class="op">=</span><span class="dv">10</span>, min_disp<span class="op">=</span><span class="fl">0.5</span>,</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>    subset<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>sc.pp.pca(nowo)</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(nowo)</span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(nowo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="48">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[:, nowo.var.highly_variable].copy()</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>sc.pp.scale(nowo, max_value<span class="op">=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="load-in-experimental-single-cell-rna-seq-data-1" class="level2" data-number="2.1">
<h2 data-number="2.1" data-anchor-id="load-in-experimental-single-cell-rna-seq-data-1"><span class="header-section-number">2.1</span> Load in experimental single cell RNA-seq data</h2>
<div class="cell" data-execution_count="49">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>alba_adata <span class="op">=</span> sc.read(<span class="st">'../data/processed/10_dataset_v2.h5ad'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="50">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>alba_adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="50">
<pre><code>AnnData object with n_obs × n_vars = 1139 × 21590
    obs: 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'n_genes', 'n_counts', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'gene_ids', 'feature_types', 'mt', 'ercc', 'ribo', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells', 'highly_variable', 'means', 'dispersions', 'dispersions_norm', 'highly_variable_nbatches', 'highly_variable_intersection', 'mean', 'std'
    uns: 'Amp_batch_ID_colors', 'S2G6+_colors', 'Stage_colors', 'dendrogram_Stage', 'hvg', 'leiden', 'leiden_colors', 'log1p', 'neighbors', 'pca', 'plate_ID_colors', 'rank_genes_groups', 'umap'
    obsm: 'X_pca', 'X_umap'
    varm: 'PCs'
    layers: 'counts'
    obsp: 'connectivities', 'distances'</code></pre>
</div>
</div>
<section id="map-experimental-data-onto-filtered-atlas" class="level3" data-number="2.1.1">
<h3 data-number="2.1.1" data-anchor-id="map-experimental-data-onto-filtered-atlas"><span class="header-section-number">2.1.1</span> Map experimental data onto filtered atlas</h3>
<div class="cell" data-execution_count="51">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>nowo.var_names <span class="op">=</span> nowo.var_names.<span class="bu">str</span>.upper()</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>alba_adata.var_names <span class="op">=</span> alba_adata.var_names.<span class="bu">str</span>.upper()</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>var_names <span class="op">=</span> nowo.var_names.intersection(alba_adata.var_names)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>nowo <span class="op">=</span> nowo[:, var_names].copy()</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>alba_adata <span class="op">=</span> alba_adata[:, var_names].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="52">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>sc.tl.ingest(alba_adata, nowo, obs <span class="op">=</span> <span class="st">'ct'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="53">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>adata_concat <span class="op">=</span> nowo.concatenate(alba_adata, batch_categories<span class="op">=</span>[<span class="st">'nowo'</span>, <span class="st">'alba'</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="54">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>adata_concat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="54">
<pre><code>AnnData object with n_obs × n_vars = 2041 × 917
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'n_counts', 'n_genes', 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'highly_variable-alba', 'means-alba', 'highly_variable_nbatches-alba', 'mean-alba', 'std-alba', 'gene_ids-alba', 'feature_types-alba', 'mt-alba', 'ercc-alba', 'ribo-alba', 'n_cells_by_counts-alba', 'mean_counts-alba', 'pct_dropout_by_counts-alba', 'total_counts-alba', 'n_cells-alba', 'dispersions-alba', 'dispersions_norm-alba', 'highly_variable_intersection-alba', 'highly_variable-nowo', 'highly_variable_rank-nowo', 'means-nowo', 'variances-nowo', 'variances_norm-nowo', 'highly_variable_nbatches-nowo', 'mean-nowo', 'std-nowo'
    obsm: 'X_pca', 'X_umap'
    layers: 'counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="55">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>alba_annotated <span class="op">=</span> adata_concat[adata_concat.obs.batch <span class="op">==</span> <span class="st">'alba'</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="56">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>alba_annotated</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="56">
<pre><code>View of AnnData object with n_obs × n_vars = 1139 × 917
    obs: 'batch', 'experiment', 'technology', 'ct', 'ct_orig', 'n_counts', 'n_genes', 'Well_ID', 'Well_coordinates', 'plate_ID', 'Subject_ID', 'Amp_batch_ID', 'Cell_barcode', 'Pool_barcode', 'Batch', 'Condition', 'SampleName', 'Stage', 'Source', 'SubGroup', 'Group', 'Clone', 'All.Events.GFP-A.Geo.Mean', 'All.Events.561D-A.Geo.Mean', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'total_counts_ercc', 'pct_counts_ercc', 'total_counts_ribo', 'pct_counts_ribo', 'leiden', 'S_score', 'G2M_score', 'phase', 'S2G6+'
    var: 'highly_variable-alba', 'means-alba', 'highly_variable_nbatches-alba', 'mean-alba', 'std-alba', 'gene_ids-alba', 'feature_types-alba', 'mt-alba', 'ercc-alba', 'ribo-alba', 'n_cells_by_counts-alba', 'mean_counts-alba', 'pct_dropout_by_counts-alba', 'total_counts-alba', 'n_cells-alba', 'dispersions-alba', 'dispersions_norm-alba', 'highly_variable_intersection-alba', 'highly_variable-nowo', 'highly_variable_rank-nowo', 'means-nowo', 'variances-nowo', 'variances_norm-nowo', 'highly_variable_nbatches-nowo', 'mean-nowo', 'std-nowo'
    obsm: 'X_pca', 'X_umap'
    layers: 'counts'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="57">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>pd.crosstab(alba_annotated.obs.Stage, alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="57">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">ct</th>
<th data-quarto-table-cell-role="th">E3.5-EPI</th>
<th data-quarto-table-cell-role="th">E3.5-ICM</th>
<th data-quarto-table-cell-role="th">E3.5-PrE</th>
<th data-quarto-table-cell-role="th">E4.5-EPI</th>
<th data-quarto-table-cell-role="th">E4.5-PrE</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">Stage</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">2iLIF</td>
<td>93</td>
<td>12</td>
<td>54</td>
<td>3</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">EPSC_GFP</td>
<td>87</td>
<td>7</td>
<td>47</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">EPSC_double</td>
<td>89</td>
<td>48</td>
<td>189</td>
<td>0</td>
<td>1</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">KOSR_GFP</td>
<td>64</td>
<td>4</td>
<td>65</td>
<td>4</td>
<td>0</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">KOSR_double</td>
<td>75</td>
<td>46</td>
<td>248</td>
<td>1</td>
<td>0</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="58">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>alba_annotated.obs.Stage.value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="58">
<pre><code>KOSR_double    370
EPSC_double    327
2iLIF          163
EPSC_GFP       142
KOSR_GFP       137
Name: Stage, dtype: int64</code></pre>
</div>
</div>
<div class="cell" data-execution_count="59">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>alba_annotated_df<span class="op">=</span>pd.DataFrame({</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">'condition'</span>:alba_annotated.obs.Stage,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">'annotation'</span>:alba_annotated.obs.ct,</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="60">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>proportions_df <span class="op">=</span> pd.crosstab(alba_annotated.obs.Stage,alba_annotated.obs.ct, normalize<span class="op">=</span><span class="st">'index'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="61">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>proportions_df_notNorm <span class="op">=</span> pd.crosstab(alba_annotated.obs.Stage,alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="62">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> proportions_df.plot.barh(</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>    stacked<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>    grid<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_11_CellProportions_Integrated_without-TE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-63-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="63">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>proportions_of_clusters_df <span class="op">=</span> pd.crosstab(alba_annotated.obs.leiden, alba_annotated.obs.ct, normalize<span class="op">=</span><span class="st">'index'</span>)</span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>proportions_of_clusters_df_notNorm <span class="op">=</span> pd.crosstab(alba_annotated.obs.leiden, alba_annotated.obs.ct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="64">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> proportions_of_clusters_df.plot.barh(</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>    stacked<span class="op">=</span><span class="va">True</span>,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>    grid<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_12_CellProportionsLeidenClusters_Integrated_without-TE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-65-output-1.png"></p>
</div>
</div>
</section>
</section>
<section id="contingency-table-analysis-1" class="level2" data-number="2.2">
<h2 data-number="2.2" data-anchor-id="contingency-table-analysis-1"><span class="header-section-number">2.2</span> Contingency table analysis</h2>
<div class="cell" data-execution_count="65">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scipy.stats <span class="im">as</span> stats</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="66">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>stats.chi2_contingency(proportions_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="66">
<pre><code>Chi2ContingencyResult(statistic=161.6138042752549, pvalue=3.931212702862735e-26, dof=16, expected_freq=array([[ 58.3880597 ,  16.74363477,  86.29411765,   1.14486392,
          0.42932397],
       [ 50.86567164,  14.58647937,  75.17647059,   0.99736611,
          0.37401229],
       [117.13432836,  33.58999122, 173.11764706,   2.29675154,
          0.86128183],
       [ 49.07462687,  14.07287094,  72.52941176,   0.96224759,
          0.36084284],
       [132.53731343,  38.00702371, 195.88235294,   2.59877085,
          0.97453907]]))</code></pre>
</div>
</div>
<div class="cell" data-execution_count="67">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>alba_chi2 <span class="op">=</span> stats.chi2_contingency(proportions_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="68">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>alba_chi2.pvalue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="68">
<pre><code>3.931212702862735e-26</code></pre>
</div>
</div>
<div class="cell" data-execution_count="69">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>alba_chi2.statistic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="69">
<pre><code>161.6138042752549</code></pre>
</div>
</div>
<div class="cell" data-execution_count="70">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># https://stackoverflow.com/questions/20453729/what-is-the-equivalent-of-r-data-chisqresiduals-in-python</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> scipy.stats.contingency <span class="im">import</span> margins</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> residuals(observed, expected):</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> (observed <span class="op">-</span> expected) <span class="op">/</span> np.sqrt(expected)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="kw">def</span> stdres(observed, expected):</span>
<span id="cb93-8"><a href="#cb93-8" aria-hidden="true" tabindex="-1"></a>    n <span class="op">=</span> observed.<span class="bu">sum</span>()</span>
<span id="cb93-9"><a href="#cb93-9" aria-hidden="true" tabindex="-1"></a>    rsum, csum <span class="op">=</span> margins(observed)</span>
<span id="cb93-10"><a href="#cb93-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># With integers, the calculation</span></span>
<span id="cb93-11"><a href="#cb93-11" aria-hidden="true" tabindex="-1"></a>    <span class="co">#     csum * rsum * (n - rsum) * (n - csum)</span></span>
<span id="cb93-12"><a href="#cb93-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># might overflow, so convert rsum and csum to floating point.</span></span>
<span id="cb93-13"><a href="#cb93-13" aria-hidden="true" tabindex="-1"></a>    rsum <span class="op">=</span> rsum.astype(np.float64)</span>
<span id="cb93-14"><a href="#cb93-14" aria-hidden="true" tabindex="-1"></a>    csum <span class="op">=</span> csum.astype(np.float64)</span>
<span id="cb93-15"><a href="#cb93-15" aria-hidden="true" tabindex="-1"></a>    v <span class="op">=</span> csum <span class="op">*</span> rsum <span class="op">*</span> (n <span class="op">-</span> rsum) <span class="op">*</span> (n <span class="op">-</span> csum) <span class="op">/</span> n<span class="op">**</span><span class="dv">3</span></span>
<span id="cb93-16"><a href="#cb93-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">return</span> (observed <span class="op">-</span> expected) <span class="op">/</span> np.sqrt(v)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="71">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(stdres(proportions_df_notNorm.to_numpy(), alba_chi2.expected_freq), columns<span class="op">=</span> proportions_df_notNorm.columns, index<span class="op">=</span>proportions_df_notNorm.index),</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span></span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb94-6"><a href="#cb94-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_13_ChiSquared_NormalisedResiduals-withoutTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-72-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="72">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(residuals(proportions_df_notNorm.to_numpy(), alba_chi2.expected_freq), columns<span class="op">=</span> proportions_df_notNorm.columns, index<span class="op">=</span>proportions_df_notNorm.index),</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_14_ChiSquared_RawResiduals-withoutTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-73-output-1.png"></p>
</div>
</div>
<section id="significance-tests-using-clusters-1" class="level3" data-number="2.2.1">
<h3 data-number="2.2.1" data-anchor-id="significance-tests-using-clusters-1"><span class="header-section-number">2.2.1</span> Significance tests using clusters</h3>
<div class="cell" data-execution_count="73">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>stats.chi2_contingency(proportions_of_clusters_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="73">
<pre><code>Chi2ContingencyResult(statistic=144.80555129331256, pvalue=6.398572999546904e-25, dof=12, expected_freq=array([[154.3880597 ,  44.27304653, 228.17647059,   3.02721686,
          1.13520632],
       [128.23880597,  36.77436348, 189.52941176,   2.51448639,
          0.9429324 ],
       [ 65.91044776,  18.90079017,  97.41176471,   1.29236172,
          0.48463565],
       [ 59.46268657,  17.05179982,  87.88235294,   1.16593503,
          0.43722564]]))</code></pre>
</div>
</div>
<div class="cell" data-execution_count="74">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2 <span class="op">=</span> stats.chi2_contingency(proportions_of_clusters_df_notNorm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="75">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2.pvalue</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="75">
<pre><code>6.398572999546904e-25</code></pre>
</div>
</div>
<div class="cell" data-execution_count="76">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a>seurat_clusters_chi2.statistic</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="76">
<pre><code>144.80555129331256</code></pre>
</div>
</div>
<div class="cell" data-execution_count="77">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(stdres(proportions_of_clusters_df_notNorm.to_numpy(), seurat_clusters_chi2.expected_freq), columns<span class="op">=</span> proportions_of_clusters_df_notNorm.columns, index<span class="op">=</span>proportions_of_clusters_df_notNorm.index),</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span>, center <span class="op">=</span> <span class="fl">0.0</span></span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_15_LeidenClusters_ChiSquared_NormalisedResiduals-withoutTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-78-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="78">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>ax <span class="op">=</span> sns.heatmap(</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>    pd.DataFrame(residuals(proportions_of_clusters_df_notNorm.to_numpy(), seurat_clusters_chi2.expected_freq), columns<span class="op">=</span> proportions_of_clusters_df_notNorm.columns, index<span class="op">=</span>proportions_of_clusters_df_notNorm.index),</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>    cmap<span class="op">=</span><span class="st">'bwr'</span>, center <span class="op">=</span> <span class="fl">0.0</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>fig <span class="op">=</span> ax.get_figure()</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>fig.savefig(<span class="ss">f'</span><span class="sc">{</span>sc<span class="sc">.</span>settings<span class="sc">.</span>figdir<span class="sc">}</span><span class="ss">/integration_16_LeidenClusters_ChiSquared_RawResiduals-withoutTE.svg'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-79-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="79">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata_concat,</span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>, <span class="st">'leiden'</span>],</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-80-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="80">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(adata_concat,</span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>],</span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a>    ncols <span class="op">=</span> <span class="dv">1</span>,</span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_17_UMAP_integrated_withoutTE.svg"</span>,</span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a> )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/umap_integration_17_UMAP_integrated_withoutTE.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-81-output-2.png"></p>
</div>
</div>
<div class="cell" data-execution_count="81">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata_concat,</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>    color<span class="op">=</span>[<span class="st">'batch'</span>, <span class="st">'ct'</span>, <span class="st">'Stage'</span>],</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>    frameon<span class="op">=</span><span class="va">False</span>,</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>    ncols<span class="op">=</span><span class="dv">1</span>,</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>    save<span class="op">=</span><span class="st">"_integration_18_PCA_integrated_withoutTE.svg"</span>,</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>WARNING: saving figure to file ../figures/pca_integration_18_PCA_integrated_withoutTE.svg</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="21_Alba_data_integration_files/figure-html/cell-82-output-2.png"></p>
</div>
</div>
</section>
</section>
</section>
<section id="session-information" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Session information</h1>
<div class="cell" data-execution_count="82">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> session_info</span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>session_info.show()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="82">

<details>
<summary>Click to view session information</summary>
<pre>-----
anndata             0.9.1
matplotlib          3.7.1
numpy               1.23.5
pandas              1.5.3
rich                NA
scanpy              1.9.3
scipy               1.10.1
seaborn             0.11.2
session_info        1.0.0
-----
</pre>
<details>
<summary>Click to view modules imported as dependencies</summary>
<pre>PIL                                         9.5.0
aa8f2297d25b4dc6fd3d98411eb3ba53823c4f42    NA
absl                                        NA
asttokens                                   NA
astunparse                                  1.6.3
attr                                        23.1.0
backcall                                    0.2.0
certifi                                     2023.05.07
cffi                                        1.15.1
charset_normalizer                          3.1.0
cloudpickle                                 2.2.1
comm                                        0.1.3
cycler                                      0.10.0
cython_runtime                              NA
dateutil                                    2.8.2
debugpy                                     1.6.7
decorator                                   5.1.1
defusedxml                                  0.7.1
etils                                       1.3.0
executing                                   1.2.0
flatbuffers                                 23.5.26
fsspec                                      2023.6.0
gast                                        NA
google                                      NA
h5py                                        3.9.0
idna                                        3.4
igraph                                      0.10.4
importlib_resources                         NA
ipykernel                                   6.23.2
jax                                         0.4.13
jaxlib                                      0.4.13
jedi                                        0.18.2
joblib                                      1.2.0
keras                                       2.12.0
kiwisolver                                  1.4.4
leidenalg                                   0.9.1
llvmlite                                    0.40.1
matplotlib_inline                           0.1.6
ml_dtypes                                   0.2.0
mpl_toolkits                                NA
mpmath                                      1.3.0
natsort                                     8.4.0
numba                                       0.57.1
numexpr                                     2.8.4
nvfuser                                     NA
opt_einsum                                  v3.3.0
packaging                                   23.1
parso                                       0.8.3
patsy                                       0.5.3
pexpect                                     4.8.0
pickleshare                                 0.7.5
pkg_resources                               NA
platformdirs                                3.8.0
plotly                                      5.15.0
prompt_toolkit                              3.0.38
psutil                                      5.9.5
ptyprocess                                  0.7.0
pure_eval                                   0.2.2
pyarrow                                     12.0.1
pycparser                                   2.21
pydev_ipython                               NA
pydevconsole                                NA
pydevd                                      2.9.5
pydevd_file_utils                           NA
pydevd_plugins                              NA
pydevd_tracing                              NA
pygments                                    2.15.1
pynndescent                                 0.5.10
pyparsing                                   3.1.0
pytz                                        2023.3
requests                                    2.31.0
setuptools                                  67.7.2
six                                         1.16.0
sklearn                                     1.2.2
skmisc                                      0.2.0
sparse                                      0.14.0
stack_data                                  0.6.2
statsmodels                                 0.14.0
sympy                                       1.12
tensorboard                                 2.12.3
tensorflow                                  2.12.0
termcolor                                   NA
texttable                                   1.6.7
threadpoolctl                               3.1.0
torch                                       2.0.1+cu117
tornado                                     6.3.2
tqdm                                        4.65.0
traitlets                                   5.9.0
typing_extensions                           NA
umap                                        0.5.3
urllib3                                     1.26.16
wcwidth                                     0.2.6
wrapt                                       1.14.1
yaml                                        6.0
zmq                                         25.1.0
zoneinfo                                    NA
</pre>
</details> <!-- seems like this ends pre, so might as well be explicit -->
<pre>-----
IPython             8.14.0
jupyter_client      8.2.0
jupyter_core        5.3.1
-----
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]
Linux-4.18.0-477.27.1.el8_8.x86_64-x86_64-with-glibc2.28
-----
Session information updated at 2023-10-23 16:34
</pre>
</details>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
});
</script>
</div> <!-- /content -->



</body></html>